neurobooth_os/iout/mbient.py

```diff
@@ -177,277 +500,176 @@ def createOutlet(self):
-        if self.buzz_time:
-            libmetawear.mbl_mw_haptic_start_motor(
-                self.device.board, 100.0, self.buzz_time
-            )
-            sleep(self.buzz_time / 1000)
+    def stop(self) -> None:
+        """Stop streaming data."""
+        self.logger.debug(self.format_message('Stopping Streaming'))
+        stop_inertial_sampling(self.device)
+        disable_inertial_sampling(self.device)
+        self.streaming = False
 
-        # print ("Acquisition started")
-        self.logger.debug(f'Mbient [{self.dev_name}]: Starting Streaming')
-        self.streaming = True
-        libmetawear.mbl_mw_acc_start(self.device.board)
-        try:  # MMRS only
-            libmetawear.mbl_mw_gyro_bmi270_start(self.device.board)
-        except:  # MMR1, MMR and MMC only
-            libmetawear.mbl_mw_gyro_bmi160_start(self.device.board)
-
-    def stop(self):
-        self.logger.debug(f'Mbient [{self.dev_name}]: Stopping Streaming...')
-        e = Event()
+    def disconnect(self) -> None:
+        """Disconnect the device."""
+        self.logger.debug(self.format_message('Disconnecting...'))
+        e = mp.Event()
         self.device.on_disconnect = lambda status: e.set()
         self.device.disconnect()
-        # libmetawear.mbl_mw_debug_reset(self.device.board)
-        print("Stopped ", self.dev_name)
-        self.streaming = False
-        e.wait(10)
-        self.logger.debug(f'Mbient [{self.dev_name}]: Streaming Stopped')
-        print(self.dev_name, self.nsmpl)
-
-    def close(self):
-        self.stop()
+        if e.wait(timeout=10):
+            self.logger.debug(self.format_message('Disconnected'))
+        else:
+            self.logger.error(self.format_message('Timed Out on Disconnect'))
 
+    def close(self) -> None:
+        """Stop streaming data, unsubscribe from data signals, and disconnect the device."""
+        try:
+            if self.streaming:
+                self.stop()
 
-def reset_mbient(mac, dev_name="mbient"):
-    # connect
-    device = MetaWear(mac)
-    device.connect()
-    print(
-        f"Connected to {device.address} {dev_name} over "
-        + ("USB" if device.usb.is_connected else "BLE")
-    )
+            for signal in self.subscribed_signals:
+                libmetawear.mbl_mw_datasignal_unsubscribe(signal)
+        except Exception as e:
+            self.logger.error(
+                self.format_message(f'Unable to stop or unsubscribe from all signals: {e}'),
+                exc_info=sys.exc_info()
+            )
+        finally:
+            self.disconnect()
 
-    # stop logging
-    libmetawear.mbl_mw_logging_stop(device.board)
-    sleep(1.0)
 
-    # flush cache if mms
-    libmetawear.mbl_mw_logging_flush_page(device.board)
-    sleep(1.0)
+# --------------------------------------------------------------------------------
+# Testing Script
+# --------------------------------------------------------------------------------
+def test_script() -> None:
```

Comment:
I agree with you in principle (and this is not necessarily pythonic). However, there is an existing paradigm for the device code held over from Neurobooth development where running the device file will run a quick test / data capture. I simply took the existing code, fixed it, and bundled it up in a function instead of leaving it as module scope. If we want to move away from this paradigm, I think we should do it for all the device files in one go.

I will add the extra comment though! (Perhaps to the argparse help description as well.)